packages/attune/src/bin/attune/main.rs

```diff
@@ -112,3 +117,108 @@ async fn main() -> ExitCode {
         ToolCommand::Apt(command) => cmd::apt::handle_apt(ctx, command).await,
     }
 }
+
+/// Infinitely retry an asynchronous function call.
+///
+/// - `operation` is the function to call.
+/// - `should_retry` evaluates whether the operation should be retried.
+/// - `retry_delay` provides the duration to wait before retrying.
+///
+/// Optionally, you can use [`retry_delay_default`] for default delay timings.
+pub async fn retry_infinite<T, E>(
+    operation: impl AsyncFn() -> Result<T, E>,
+    should_retry: impl Fn(&E) -> bool,
+    retry_delay: impl Fn(usize) -> Duration,
+) -> Result<T, E> {
+    for attempt in 0usize.. {
+        match operation().await {
+            Ok(value) => return Ok(value),
+            Err(e) => {
+                if should_retry(&e) {
+                    tokio::time::sleep(retry_delay(attempt)).await;
+                } else {
+                    return Err(e);
+                }
+            }
+        }
+    }
+    unreachable!("loop is functionally infinite");
+}
+
+/// The default retry delay is a static delay of 2 seconds
+/// plus a random jitter of up to 2 seconds.
+pub fn retry_delay_default(_: usize) -> Duration {
+    const STATIC_RETRY_DELAY_MS: u64 = 2000;
+    Duration::from_millis(STATIC_RETRY_DELAY_MS + rand::random_range(0..STATIC_RETRY_DELAY_MS))
+}
+
+/// The result of signing content with a GPG key.
+#[derive(Debug, Clone)]
+pub struct SignedGpgContent {
+    pub clearsigned: String,
+    pub detachsigned: String,
+    pub public_key_cert: String,
+}
+
+/// Sign content with the named GPG key ID.
+pub async fn gpg_sign(
+    gpg_home_dir: Option<impl Into<String>>,
+    key_id: impl Into<String>,
+    content: impl Into<Vec<u8>>,
+) -> Result<SignedGpgContent> {
+    let gpg_home = gpg_home_dir.map(|p| p.into());
+    let key_id = key_id.into();
+    let content = content.into();
+    tokio::task::spawn_blocking(move || gpg_sign_blocking(gpg_home, key_id, content))
```

Comment:
[q] Why? There's not really any place where we don't immediately need to wait on this anyway.